scripts/gas-refund-program/staking/index.ts

```diff
@@ -1,11 +1,22 @@
 import BigNumber from 'bignumber.js';
+import { BlockInfo } from '../../../src/lib/block-info';
+import { CHAIN_ID_MAINNET } from '../../../src/lib/constants';
 import { GRP_MIN_STAKE } from '../../../src/lib/gas-refund';
 import { StakedPSPByAddress } from '../types';
 import { getSPSPStakes } from './spsp-stakes';
 
 // @TODO: fetch safety-module stakes
-export const getPSPStakes = async (): Promise<StakedPSPByAddress | null> => {
-  const stakesByAddress = await getSPSPStakes();
+export const getPSPStakes = async (
+  epochEndTime: number,
+): Promise<StakedPSPByAddress | null> => {
+  const blockNumber =
+    Date.now() > epochEndTime * 1000
+      ? (await BlockInfo.getInstance(CHAIN_ID_MAINNET).getBlockAfterTimeStamp(
```

Comment:
Since this gets block after the timestamp, can this potentially get the 1st block of the next period. And in the next period PSP to sPSP rate can be different
should we `blockNumber--` just to be safe?

Reply:
as discussed, we can cover it later with all things accounted